src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5231,6 +5250,15 @@ public object VisitBinaryExpression(BinaryExpressionAst binaryExpressionAst)
                         CachedReflectionInfo.ParserOps_SplitOperator,
                         _executionContextParameter, Expression.Constant(binaryExpressionAst.ErrorPosition), lhs.Cast(typeof(object)), rhs.Cast(typeof(object)),
                         ExpressionCache.Constant(false));
+                case TokenKind.QuestionQuestion when ExperimentalFeature.IsEnabled("PSNullCoalescingOperators") :
+                    if (lhs is ConstantExpression lhsConstExpr && lhsConstExpr.Value != null)
+                    {
+                        return lhs;
+                    }
+                    else
+                    {
+                        return Expression.Coalesce(lhs, rhs);
```

Comment:
This should use either [`IsNull` or `IsNullLike`](https://github.com/PowerShell/PowerShell/blob/eb2220632cf43d40273844ff98b162e704286900/src/System.Management.Automation/engine/LanguagePrimitives.cs#L1061-L1074) from `LanguagePrimitives`.

Reply:
Definitely should be `IsNullLike` unless we want these operators to treat DBNull / NullString as "not null". 🙂

Reply:
Changed to use IsNullLike